analytics.ts

```diff
@@ -68,10 +68,15 @@ export const inject = (deps: UsageAnalyticsDeps) => {
 					return taqResolve('');
 				}
 			}),
-			mapRej(() =>
-				option === OPT_IN
-					? 'The command "taq opt-in" is ignored as this might be the first time running Taqueria...'
-					: 'The command "taq opt-out" is ignored as this might be the first time running Taqueria...'
+			mapRej(previous =>
```

Comment:
Do we still need it? I thought this would not be needed after the internal task refactor

Reply:
I'm not sure to be honest. If not, I assume we'll address that in a different ticket/PR. 
The acceptance criteria for this particular issue/PR was:

>  Refactor remaining internal tasks so that the execution flow matches that of plugin-provided tasks

As that criteria is a little vague, I interpreted it to be:

> Refactor tasks to not fork but run in the main execution pipeline

The motive for that , as far as I was aware, was to resolve a) memory leaks; b) concurrency issues; c) make the code easier to reason about; d) ability to ask for `--help` for built-in tasks

I had thought that the only thing holding back GA tracking was issue B above.

--
I modified the following code strictly for type correctness, as this section of code was returning Future<string, unknown> instead of Future<TaqError, unknown>:

Before:
```
mapRej(() =>
  option === OPT_IN
	  ? 'The command "taq opt-in" is ignored as this might be the first time running Taqueria...'
	  : 'The command "taq opt-out" is ignored as this might be the first time running Taqueria...'
)
```

After:
```
mapRej(previous => TaqError.create({
	kind: 'E_OPT_IN_WARNING',
	msg: option === OPT_IN
		? 'The command "taq opt-in" is ignored as this might be the first time running Taqueria...'
		: 'The command "taq opt-out" is ignored as this might be the first time running Taqueria...',
	previous,
	context: option,
})
```

If the above code isn't needed anymore, great - but I'd probably prefer making that adjustment in a different PR, as I cannot recall the reason why Jimmy added this code, and too, I think that's outside of the scope I had in mind in this PR. Would you be fine if we created an issue for this instead, Ali? I would do so, but given that you recognized this, I assume you might have more context/recollection about why this code was here to start with.